### PR TITLE
mix of improvement

### PR DESCRIPTION
**Pull request by pandaro**
convert ai. in self.ai. for:
assist system
attack system
raider system
bomber system

### CODE REVIEW

File: assistbehaviour.lua [hunk 1]
```diff
@@ -56,15 +56,15 @@ function AssistBehaviour:Update()
 		local uname = self.name
 		if self.isCommander then
 			-- turn commander into build assister if you control more than half the mexes or if it's damaged
-			if ai.nonAssistant[self.id] then
-				if (IsSiegeEquipmentNeeded() or unit:GetHealth() < unit:GetMaxHealth() * 0.9) and ai.factories ~= 0 and ai.conCount > 2 then
-					ai.nonAssistant[self.id] = nil
+			if self.ai.nonAssistant[self.id] then
+				if (self.ai.haveAdvFactory or IsSiegeEquipmentNeeded() or unit:GetHealth() < unit:GetMaxHealth() * 0.9) and self.ai.factories ~= 0 and self.ai.conCount > 2 then
```
Comment (eronoobos) [inline assistbehaviour.lua]: i would prefer not using functions from taskqueues in a module. i'm not even sure how taskqueues is available here

**Comment by pandaro** [comment]
retracted

File: taskqueues.lua [hunk 2]
```diff
@@ -674,6 +674,7 @@ local anyLvl2ShipYard = {
 	Lvl2ShipBreakthrough,
 	Lvl2ShipMerl,
 	Lvl2ShipAssist,
+	Lv2SubWar,
```
Comment (eronoobos) [inline taskqueues.lua]: it's easier to find things in the future if the naming conventions are consistent. function should be named `Lvl2SubWar`

**Comment by pandaro** [comment]
sorry, i remember you correct me anytime in the past from lvl2 to lv2
fix in a moment

**Pull request merged**